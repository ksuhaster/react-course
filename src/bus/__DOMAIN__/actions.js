// Types
import { types } from './types';

export const domainActions = Object.freeze({
    domainAction: () => {
        return {
            type: types.DOMAIN_ACTION
        }
    }
});