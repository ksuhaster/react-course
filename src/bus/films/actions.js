// Types
import { types } from './types';

import { api } from '../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());

        const response = await api.films.fetch();

        if (response.status === 200) {
            const { results } = await response.json();
            dispatch(filmsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmsActions.setFetchingError(error));
        }

        dispatch(filmsActions.stopFetching());
    }
});


export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());

        const response = await api.film.fetch(id);

        if (response.status === 200) {
            const results = await response.json();
            dispatch(filmActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmActions.setFetchingError(error));
        }

        dispatch(filmActions.stopFetching());
    }
});