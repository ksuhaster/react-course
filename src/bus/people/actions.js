// Types
import { types } from './types';

import { api } from '../../api';

export const peopleActions = Object.freeze({
    // Sync
    fill: (payload) => {
        return {
            type: types.PEOPLE_FILL,
            payload
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.PEOPLE_FETCH_ASYNC
        });

        const response = await api.people.fetch();
        const { results } = await response.json();

        dispatch(peopleActions.fill(results));
    }
});