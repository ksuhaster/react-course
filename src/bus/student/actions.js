// Types
import { types } from './types';

export const studentActions = Object.freeze({
    setStudent: (payload) => {
        return {
            type: types.SET_NAME,
            payload
        }
    }
});