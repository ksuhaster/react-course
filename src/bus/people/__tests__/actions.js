import { peopleActions, personActions} from "../actions";
import expectExport from "expect";

describe("people list actions:", () => {
    // Sync
    test("startFetching", () => {
        expectExport(peopleActions.startFetching()).toMatchSnapshot();
    });
    test("stopFetching", () => {
        expectExport(peopleActions.stopFetching()).toMatchInlineSnapshot(`
      Object {
        "type": "PEOPLE_STOP_FETCHING",
      }
    `);
    });
    test("fill", () => {
        expectExport(peopleActions.fill()).toMatchSnapshot();
    });
    test("setFetchingError", () => {
        expectExport(peopleActions.fill()).toMatchSnapshot();
    });

    // Async
    test("fetchAsync", () => {
        expectExport(peopleActions.fetchAsync()).toMatchSnapshot();
    });
});


describe("one person actions:", () => {
    // Sync
    test("startFetching", () => {
        expectExport(personActions.startFetching()).toMatchSnapshot();
    });
    test("stopFetching", () => {
        expectExport(personActions.stopFetching()).toMatchInlineSnapshot(`
      Object {
        "type": "PERSON_STOP_FETCHING",
      }
    `);
    });
    test("fill", () => {
        expectExport(personActions.fill()).toMatchSnapshot();
    });
    test("setFetchingError", () => {
        expectExport(personActions.fill()).toMatchSnapshot();
    });

    // Async
    test("fetchAsync", () => {
        expectExport(personActions.fetchAsync()).toMatchSnapshot();
    });
});
