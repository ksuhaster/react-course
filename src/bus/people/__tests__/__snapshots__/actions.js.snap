// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`one person actions: fetchAsync 1`] = `[Function]`;

exports[`one person actions: fill 1`] = `
Object {
  "payload": undefined,
  "type": "PERSON_FILL",
}
`;

exports[`one person actions: setFetchingError 1`] = `
Object {
  "payload": undefined,
  "type": "PERSON_FILL",
}
`;

exports[`one person actions: startFetching 1`] = `
Object {
  "type": "PERSON_START_FETCHING",
}
`;

exports[`people list actions: fetchAsync 1`] = `[Function]`;

exports[`people list actions: fill 1`] = `
Object {
  "payload": undefined,
  "type": "PEOPLE_FILL",
}
`;

exports[`people list actions: setFetchingError 1`] = `
Object {
  "payload": undefined,
  "type": "PEOPLE_FILL",
}
`;

exports[`people list actions: startFetching 1`] = `
Object {
  "type": "PEOPLE_START_FETCHING",
}
`;
